tests/tst_ncio_mpi.F90

```diff
@@ -0,0 +1,196 @@
+  quantize1(5,:) = 123.456789
+  quantize1(10,:) = 1013.254321
+  call quantize_data(quantize1, quantize2, nbits, qerr)
+  if (abs(quantize2(1,1)-quantize1(1,1)) .gt. qerr) stop 10
+
+
+  if (my_rank .eq. 0) print *,'*** Test quantize data to 32 bit ...'
+  nbits = 32
+  call quantize_data(quantize1, quantize2, nbits, qerr)
+  if (abs(quantize2(1,1)-quantize1(1,1)) .gt. qerr) stop 11
+
+
+  if (my_rank .eq. 0) print *,'*** Test write of attributes...'
+  call write_attribute(dset,'bar',values_1d,'ugrd')
+  call write_attribute(dset,'hello','world')
+
+
+  if (my_rank .eq. 0) print *,'*** Test read of attribute just written...'
+  call read_attribute(dset,'hello',charatt)
+  if (trim(charatt) .ne. 'world') stop 12
+
+
+  !!!!!!!
+  call read_attribute(dset,'missing_value',mval,'pressfc')
+
+  if (my_rank .eq. 0) print *,'*** Test read of missing_value attribute...'
+  if (mval .ne. -1.e10) stop 13
+
+  nvar = get_nvar(dset, 'vgrd')
+
+  if (my_rank .eq. 0) print *,'*** Test getting variable id...'
+  if (trim(dset%variables(nvar)%name) .ne. 'vgrd') stop 14
+
+
+  if (my_rank .eq. 0) print *,'*** Test variable dimension info...'
+  if (trim(adjustl(dset%variables(nvar)%dimnames(1))) .ne. 'grid_xt' .or. &
+       dset%variables(nvar)%dimlens(1) .ne. 256 .or. &
+       dset%dimensions(dset%variables(nvar)%dimindxs(1))%len .ne. 256) stop 15
+
+  if (trim(adjustl(dset%variables(nvar)%dimnames(2))) .ne. 'grid_yt' .or. &
+       dset%variables(nvar)%dimlens(2) .ne. 128 .or. &
+       dset%dimensions(dset%variables(nvar)%dimindxs(2))%len .ne. 128) stop 16
+
+  if (trim(adjustl(dset%variables(nvar)%dimnames(3))) .ne. 'pfull' .or. &
+       dset%variables(nvar)%dimlens(3) .ne. 64 .or. &
+       dset%dimensions(dset%variables(nvar)%dimindxs(3))%len .ne. 64) stop 17
+
+  if (trim(adjustl(dset%variables(nvar)%dimnames(4))) .ne. 'time' .or. &
+       dset%variables(nvar)%dimlens(4) .ne. 1 .or. &
+       dset%dimensions(dset%variables(nvar)%dimindxs(4))%len .ne. 1) stop 18
+
+  if (my_rank .eq. 0) print *,'*** Test getting idate from time units attribute...'
+  idate = get_idate_from_time_units(dset)
+  icheck = (/2016,1,4,6,0,0/)
+  if (all(idate .ne. icheck)) then
+     if (my_rank .eq. 0) print *,'*** idate not correct'
+     stop 19
+  endif
+
+  if (my_rank .eq. 0) print *,'*** Test getting time units from idate...'
+  time_units = get_time_units_from_idate(idate, time_measure='minutes')
+  if (trim(time_units) .ne. 'minutes since 2016-01-04 06:00:00') then
+     if (my_rank .eq. 0) print *,'***time units not correct...'
+     stop 20
+  endif
+
+  if (my_rank .eq. 0) print *,'*** Test the return code is correct for missing attribute...'
+  call read_attribute(dset,'foo',time_units,errcode=ierr)
+  if (ierr .ne. NF90_ENOTATT) then
+     if (my_rank .eq. 0) print *,'***return code not correct for missing attribute...'
+     stop 21
+  endif
+  call close_dataset(dset)
+  !!!!!!!!
+
+  ! re-open dataset
+  dset = open_dataset('dynf000.nc')
```

Comment:
This is an MPI test, so shouldn't this be with the following args `paropen=.true., mpicomm=MPI_COMM_WORLD`

Reply:
yes, fixed